doc/sphinx/addendum/type-classes.rst

```diff
@@ -521,6 +559,20 @@ Settings
    the dependent ones previously (Coq 8.5 and below). This can result in
    quite different performance behaviors of proof search.
 
+   .. _TypeclassesDefaultMode:
+
+.. opt:: Typeclasses Default Mode {? ( {| "+" | "-" | "!" } ) }.
+
+   This option (unset by default) controls the default mode declaration
+   associated to a :cmd:`Class`. If set, the each class declaration uses
+   this default mode for *all* its indices, unless an :attr:`mode` attribute
+   is used to set the mode explicitly.
+
+   .. warn:: Using inferred default mode declaration “mode” for “@ident”
```

Comment:
The actual warning message seems to rather be "Using an inferred default mode: ..."

Also, its documentation should indicate more clearly that the warning is disabled by default and that its name is "class-declaration-default-mode" with a link to the documentation of the warning machinery (so that users know how to turn it on).

Reply:
That's what we had before. I'll add the link.